### PR TITLE
10 status updates component

### LINKED ISSUE #10: Status updates component
## Description
As a purchaser I want a page that shows me the current status because I want to see relevant status updates.

This is the page that is first shown to the user (in the future this is only shown when logged in).

## Acceptance criteria
- [x] Status updates are shown 
- [x] Status updates are fetched from the backend 

## Dependencies

### PR DESCRIPTION
**Pull request by martinjonsson01**
## Summary
Adds status update notifications that are fetched from the backend and shown on the dashboard. 

image

## Definition of done
- [x] All acceptance criteria are completed.
- [x] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [ ] The code is reviewed and accepted by (at least) two other team members.
- [x] Views adhere to design style.

Closes #10

### CODE REVIEW

File: src/Client/Pages/Contracts/RecentlyViewed.razor [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    /// <summary>
+    /// Tells the component to render itself again.
+    /// </summary>
+    public async void AddViewedContract(Contract newRecent)
+    {
+        _recent.Add(newRecent);
+        await InvokeAsync(StateHasChanged);
+    }
+
+}
```
Comment (XwayTomy) [inline src/Client/Pages/Contracts/RecentlyViewed.razor:55]: A bit weird to see these changes in this PR, but makes sense after reading commit history.

File: src/Client/Pages/Dashboard/DateTimeFormattingExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/// <summary>
+/// Extensions for <see cref="DateTime"/> to format it in different ways.
+/// </summary>
+internal static class DateTimeFormattingExtensions
+{
+    private static readonly SortedList<double, Func<TimeSpan, string>> _offsets = new()
+    {
+        // Offset in minutes, lambda that formats value as string
+        // F0 converts floating point numbers like 6.789 into 6
```
Comment (XwayTomy) [inline src/Client/Pages/Dashboard/DateTimeFormattingExtensions.cs:11]: Good documentation!

File: src/Client/Shared/NavMenu.razor [hunk 3]
```diff
@@ -8,13 +8,13 @@
...
-                        <span class="oi oi-list-rich" aria-hidden="true"></span> Dashboard
-                    </NavLink>
-                </div>
         <div class="nav-item px-3">
             <NavLink class="nav-link" href="" Match="NavLinkMatch.All">
+                <span class="oi oi-list-rich" aria-hidden="true"></span> Dashboard
+            </NavLink>
+        </div>
+        <div class="nav-item px-3">
+            <NavLink class="nav-link" href="contracts">
```
Comment (XwayTomy) [inline src/Client/Shared/NavMenu.razor:17]: What happens here?

**Review by XwayTomy — APPROVED** [review]
Good job with this one!

File: src/Client/Pages/Contracts/RecentlyViewed.razor (hunk 1 above)
Comment (martinjonsson01) [reply src/Client/Pages/Contracts/RecentlyViewed.razor:55]: Yeah, moving files makes the PRs look weird sometimes.

File: src/Client/Shared/NavMenu.razor (hunk 3 above)
Comment (martinjonsson01) [reply src/Client/Shared/NavMenu.razor:17]: It wasn't indented correctly 😩

File: src/Client/Pages/Admin/ContractList.razor [hunk 4]
```diff
@@ -1,41 +1,31 @@
...
-        <em>Loading...</em>
-    </p>
-}
-else
-{
+<FetchData @ref="_dataFetcher"
+           TData="List<Contract>"
+           Url="/api/v1/contracts"
+           Context="contracts">
+
```
Comment (JacobBredin) [inline src/Client/Pages/Admin/ContractList.razor:8]: Nice 👍

**Review by JacobBredin — APPROVED** [review]
LGTM

**Pull request merged**